keps/sig-cluster-lifecycle/0014-20180707-componentconfig-api-types-to-staging.md

```diff
@@ -0,0 +1,296 @@
+---
+kep-number: 14
+title: Moving ComponentConfig API types to staging repos
+status: provisional
+authors:
+  - "@luxas"
+  - "@sttts"
+owning-sig: sig-cluster-lifecycle
+participating-sigs:
+  - sig-api-machinery
+  - sig-node
+  - sig-network
+  - sig-scheduling
+  - sig-cloud-provider
+reviewers:
+  - "@thockin"
+  - "@liggitt"
+  - "@wojtek-t"
+  - "@stewart-yu"
+  - "@dixudx"
+approvers:
+  - "@thockin"
+  - "@jbeda"
+editor:
+  name: "@luxas"
+creation-date: 2018-07-07
+last-updated: 2018-07-07
+---
+
+# Moving ComponentConfig API types to staging repos
+
+**How we can start supporting reading versioned configuration for all our components after a code move for ComponentConfig to staging**
+
+## Table of Contents
+
+ * [Moving ComponentConfig API types to staging repos](#moving-componentconfig-api-types-to-staging-repos)
+    * [Table of Contents](#table-of-contents)
+    * [Summary](#summary)
+       * [The current state of the world](#the-current-state-of-the-world)
+          * [Current kubelet](#current-kubelet)
+          * [Current kube-proxy](#current-kube-proxy)
+          * [Current kube-scheduler](#current-kube-scheduler)
+          * [Current kube-controller-manager](#current-kube-controller-manager)
+          * [Current kube-apiserver](#current-kube-apiserver)
+          * [Current cloud-controller-manager](#current-cloud-controller-manager)
+       * [Goals](#goals)
+       * [Non-goals](#non-goals)
+       * [Related proposals and further references](#related-proposals-and-further-references)
+    * [Proposal](#proposal)
+       * [Migration strategy per component or k8s.io repo](#migration-strategy-per-component-or-k8sio-repo)
+          * [k8s.io/apimachinery changes](#k8sioapimachinery-changes)
+          * [k8s.io/apiserver changes](#k8sioapiserver-changes)
+          * [kubelet changes](#kubelet-changes)
+          * [kube-proxy changes](#kube-proxy-changes)
+          * [kube-scheduler changes](#kube-scheduler-changes)
+          * [k8s.io/controller-manager changes](#k8siocontroller-manager-changes)
+          * [kube-controller-manager changes](#kube-controller-manager-changes)
+          * [cloud-controller-manager changes](#cloud-controller-manager-changes)
+          * [kube-apiserver changes](#kube-apiserver-changes)
+       * [Timeframe and Implementation Order](#timeframe-and-implementation-order)
+       * [OWNERS files for new packages and repos](#owners-files-for-new-packages-and-repos)
+
+## Summary
+
+Currently all ComponentConfiguration API types are in the core Kubernetes repo, which make them practically inaccessible to for any third-party tool,
+and blocks e.g. kubeadm to move out of the core repo. This KEP is about creating new staging repos, `k8s.io/{component}`, which will host the external
```

Comment:
Two things.

1) A nit. The first sentence is hard to read. It might be wroth breaking into 2 or 3 sentences.
2) Given that k8s 1.11 makes it harder to vendor (missing generated files) this may be more important now than before.

Reply:
1. done 2. done